papaya-core/src/main/java/pl/edu/agh/papaya/security/KeycloakProvider.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.agh.papaya.security;
+
+import lombok.RequiredArgsConstructor;
+import org.keycloak.adapters.springboot.KeycloakSpringBootProperties;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class KeycloakProvider {
+
+    private final KeycloakClientConfiguration config;
+
+    private final KeycloakSpringBootProperties props;
+
+    // it needs to be populated lazily because
+    // in tests we override the configuration above
+    // and somehow it doesn't work
+    @SuppressWarnings("PMD.AvoidUsingVolatile")
+    private transient volatile Keycloak keycloak;
+
+    public RealmResource getRealm() {
+        return getKeycloak().realm(props.getRealm());
+    }
+
+    public Keycloak getKeycloak() {
+        initialize();
+        return keycloak;
+    }
+
+    private void initialize() {
+        if (keycloak == null) {
```

Comment:
Z szacunku do tabulatora niby można 
```
if (keycloak != null) {
    return;
} else {
    // ...
}
```
To dużo wcięć jest

Reply:
wyciągnąłem do metody, wygląda lepiej